x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/inference_rescore.yml

```diff
@@ -0,0 +1,193 @@
+                            "right_child": 2
+                          },
+                          {
+                            "node_index": 1,
+                            "leaf_value": 5.0
+                          },
+                          {
+                            "node_index": 2,
+                            "leaf_value": 2.0
+                          }
+                        ],
+                        "target_type": "regression"
+                      }
+                    },
+                    {
+                      "tree": {
+                        "feature_names": [
+                          "type_tv"
+                        ],
+                        "tree_structure": [
+                          {
+                            "node_index": 0,
+                            "split_feature": 0,
+                            "split_gain": 12,
+                            "threshold": 1,
+                            "decision_type": "lt",
+                            "default_left": true,
+                            "left_child": 1,
+                            "right_child": 2
+                          },
+                          {
+                            "node_index": 1,
+                            "leaf_value": 1.0
+                          },
+                          {
+                            "node_index": 2,
+                            "leaf_value": 12.0
+                          }
+                        ],
+                        "target_type": "regression"
+                      }
+                    }
+                  ]
+                }
+              }
+            }
+          }
+
+  - do:
+      headers:
+        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
+      indices.create:
+        index: store
+        body:
+          mappings:
+            properties:
+              product:
+                type: keyword
+              cost:
+                type: integer
+
+  - do:
+      headers:
+        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
+        Content-Type: application/json
+      bulk:
+        index: store
+        refresh: true
+        body: |
+          { "index": {} }
+          { "product": "TV", "cost": 300 }
+          { "index": {} }
+          { "product": "TV", "cost": 400}
+          { "index": {} }
+          { "product": "TV", "cost": 600}
+          { "index": {} }
+          { "product": "VCR", "cost": 15}
+          { "index": {} }
+          { "product": "VCR", "cost": 350}
+          { "index": {} }
+          { "product": "VCR", "cost": 580}
+          { "index": {} }
+          { "product": "Laptop", "cost": 100}
+          { "index": {} }
+          { "product": "Laptop", "cost": 300}
+          { "index": {} }
+          { "product": "Laptop", "cost": 500}
+
+---
```

Comment:
Great tests! But let's add more tests with different values of `window_size` and also combining scores from a query and LTR

Reply:
100%! I am adding them now :)